### PR TITLE
Add Catalina and github actions support 

### PR DESCRIPTION
**Pull request by Stevemoretz**
This is a good starting point, I even got it build artifacts but for creating releases it gets an error.

I'm not really a fan of github and work on gitlab so couldn't figure out why it fails to create releases yet, maybe you can test it and find out?

It build both arm and amd as artifacts for now, you might wanna try it and download and see if the build works fine for you, I still haven't downloaded and tried it myself.

Basically a draft PR for now, so we can test and figure things out.

### CODE REVIEW

**Comment by Stevemoretz** [comment]
I'm trying a bit more to resolve the error of Github Release.

**Comment by Stevemoretz** [comment]
And in case you were wondering what was the change that allowed 10.15 to work too, it's just a simple env variable, which Xcode uses:  `MACOSX_DEPLOYMENT_TARGET: 10.15`

You can easily set that in your own terminal too.

```bash
export  MACOSX_DEPLOYMENT_TARGET: 10.15
bash nekoray_macos_builder.sh
```

or 

```bash
MACOSX_DEPLOYMENT_TARGET: 10.15 bash nekoray_macos_builder.sh
```

**Comment by aaaamirabbas** [comment]
Hi, thank you very much for the Pull Request (GitHub Actions is a good idea).

There are several items:

- This script is for building on macOS, and for example, if the script is running within the Ubuntu, it becomes incompatible and the wrong version will be set, so flags must be set up (I have planned for this)
- 
- The next issue is why did you change the clone from SSH to https? Is there any particular reason? If there is no specific reason, please return the code to the previous state.

**I added flag MACOSX_DEPLOYMENT_TARGET into the script and rebuild again source, please re -download and test the latest version.**

If the problem is fixed, then MACOSX_DEPLOYMENT_TARGET will work properly and I thank you.

**Comment by Stevemoretz** [comment]
Thank you for sharing your concerns.

> This script is for building on macOS, and for example, if the script is running within the Ubuntu, it becomes incompatible and the wrong version will be set, so flags must be set up (I don't have planned for this)

Yes that's already covered by `runs-on: macos-latest` in the workflow file, it runs on the latest version of MacOS and the files it generates, work on MacOS at the end, maybe you are talking about the create_release step of the workflow which is on ubuntu? That step is only for releasing the build is already finished in the last step so it doesn't matter what you use in the release step, I used MacOS for build, and Linux for release that's the beauty of working with CI.

> The next issue is why did you change the clone from SSH to https? Is there any particular reason? If there is no specific reason, please return the code to the previous state.

SSH = authentication, you wouldn't want to create a SSH-key and do a lot of extra steps in the CI every time, you would do it locally because it's just once, but using it in CI means a lot more work and at the end the build are slower because of all the extra unnecessary steps, that's why it's almost always better to use https for `git clone` on CI.

I don't have access to my Mac right now, but I could finally get the CI working, I might add some stuff to it but for now it built and gave the release : https://github.com/Stevemoretz/nekoray-macos/releases/tag/2.25

If you have access to your mac, could you get the ARM version and test it?

**Comment by aaaamirabbas** [comment]
My laptop is not an ARM version and I don't have access to Apple's CPUs.

**Comment by Stevemoretz** [comment]
> My laptop is not an ARM version and I don't have access to Apple's CPUs.

Oh! I thought you had ARM sorry it's okay though you can still test the AMD64 if you want it's in the releases page as well.

**Comment by aaaamirabbas** [comment]
Hi, I merged source, but build is failed, the reason for failure is that the 2.25 tag is already available (this requires a fix so that each Build is made a new tag at the Build number)

Also set MACOSX_DEPLOYMENT_TARGET to **10.9** and add to the main script. (The minimum version that is supported is 10.9)
This is useful for people who want to make their own version and that more people can use.

**Comment by Stevemoretz** [comment]
> Hi, I merged source, but build is failed, the reason for failure is that the 2.25 tag is already available (this requires a fix so that each Build is made a new tag at the Build number)
> 
> Also set MACOSX_DEPLOYMENT_TARGET to **10.9** and add to the main script. (The minimum version that is supported is 10.9) This is useful for people who want to make their own version and that more people can use.

Let's discuss this a bit further before we rush into it, your script builds the latest tag of the https://github.com/MatsuriDayo/nekoray my action also gets that latest tag and sets it at the end, since you are not modifying the source code of https://github.com/MatsuriDayo/nekoray in any way in your builds, it makes sense to have only one tag for each tag of the source repository, that makes it cleaner and syncs it better with the original repository.

We could go ahead and do stuff like 2.25-2023-01-01 but that sounds a bit unnecessary at least that's how I see it, since the source code isn't changed at all, and even if there is a bug in the build script it's better to remove that faulty build from the releases section and let CI regenerate it.

Another advantage of using the same tag as the source repository is the next step of this PR, which I was gonna provide in a separate PR which I'm working on right now: https://github.com/Stevemoretz/nekoray-macos/blob/main/.github/workflows/blank.yml

That's the ability to have automatic builds from the main repository whenever they push a tag (every night at 12.00 the CI would go check that main repo and if there isn't any newer tags then it builds the latest version)

So I would suggest not to mess with the tags to allow these kind of functionality, however if you insist on that, I could stop working on this automatic build and provide a simple date to have multiple tags.

---------------

> Also set MACOSX_DEPLOYMENT_TARGET to 10.9 and add to the main script. (The minimum version that is supported is 10.9)

That's a great idea, since you closed this pull request I have to do that in another pull request, but make sure to read my comment about the tags, so I can provide a PR that solves both issues.

**Comment by Stevemoretz** [comment]
@aaaamirabbas Here's your second request: 
https://github.com/Stevemoretz/nekoray-macos/commit/7919fbaed7b47defc945aa2815efddbc478d502c

CI built another version, if you have access to your Mac system you could try it to make sure it still runs fine for you: 
https://github.com/Stevemoretz/nekoray-macos/releases/tag/2.25

(I'll provide the PR later as I stated earlier)

**Comment by Stevemoretz** [comment]
@aaaamirabbas 
The automatic build for every night is also ready, so you can see how it works where there is a new tag available:
https://github.com/Stevemoretz/nekoray-macos/actions/runs/4723476018/jobs/8379478797

And how it just skips the build when there is no newer version available:
https://github.com/Stevemoretz/nekoray-macos/actions/runs/4723476018/jobs/8380285519

```
Latest tag on MatsuriDayo/nekoray: 2.25
Latest tag on this repository: 2.25
Latest tags match! Aborting workflow.
```

**Comment by Stevemoretz** [comment]
Since I finished it, I created the pull request too, we can talk more about these there: https://github.com/aaaamirabbas/nekoray-macos/pull/7
